### PR TITLE
Speed optimization

### PR DESCRIPTION
**Pull request by Haolicopter**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,11 +22,11 @@ export default function patchAxios(axios, vue, options) {
    *  - added to the `pendingRequests` hash if not, with a cancel function
    */
   axios.interceptors.request.use((config) => {
-    const { requestId } = config;
+    const { requestId, context } = config;
     if (requestId) {
       const source = CancelToken.source();
```
Comment (Bellardia) [inline index.js]: If `requestId` is not passed by context is, generate a random `requestId` and assign it to the context

Comment (Haolicopter) [reply index.js]: I guess we could add this, but request id signals which component its coming from and which method is making the request. Adding request id and context is an intention behaviour. Not sure if we should encourage request omit any of the two

Comment (Bellardia) [reply index.js]: The use case was mentioned earlier, but at times you may want a request to have duplicates or to not die with the component. Allowing either, both, or none maintains full flexibility on the caller side.

ex: I want to cancel duplicate requests but keep them alive when component dies or
ex: I want to allow duplicate requests but kill them when component dies

Comment (Haolicopter) [reply index.js]: The current logic can't handle this flexibility. Context is strictly mapped with requests.

Comment (Haolicopter) [reply index.js]: `this.cancelRequest` needs context, thats why I used the contextMapping as outer loop

Comment (Haolicopter) [reply index.js]: If you just extract context from requestId, then it is still strictly related.

Comment (Bellardia) [reply index.js]: Yeah but there's nothing forcing you to do it that way: re "The current logic can't handle this flexibility. Context is strictly mapped with requests." 

It's quite simple to rewrite to remove the limitation which is my point above

**Pull request merged**